### ISSUE TITLE
[dev branch] frontend https listeners not working with existing tls secrets

### ISSUE THREAD

**Issue by bithavoc** [haproxytech/kubernetes-ingress#116]
In the `dev` branch, the controller never enables the https frontend listeners for existing ingresses with TLS secrets:

```
	if len(certsUsed) > 0 {
		usingHTTPS = true
		c.enableCerts() // HACK: force enable certs on every iteration
	}
```

Also, `handleDefaultCertificate` may change `c.UseHTTPS` but because enableCerts is evaluated **before** `handleDefaultCertificate` then it may never enable https listeners for default cert:

Snippet for reference:
```
	if c.UseHTTPS.Status != EMPTY {
		c.enableCerts()
		c.UseHTTPS.Status = EMPTY
	}

	usingHTTPS := false
	reload = c.handleDefaultCertificate(certsUsed)
	needsReload = needsReload || reload
	if len(certsUsed) > 0 {
		usingHTTPS = true
	}
```

/cc @Mo3m3n

**Comment by Mo3m3n** [comment]
Thank you @bithavoc for reporting this.
The new dev image should fix this and today we will issue a 1.2.5 release which will merge all the bug fixed in dev.

**Comment by bithavoc** [comment]
Ok cool trying the new image
